lua/starfall/libs_sh/surfaceinfo.lua

```diff
@@ -0,0 +1,79 @@
+-- Globals
+local checkluatype = SF.CheckLuaType
+local checkpermission = SF.Permissions.check
+local registerprivilege = SF.Permissions.registerPrivilege
+
+
+--- SurfaceInfo type
+-- @name SurfaceInfo
+-- @class type
+-- @libtbl surfaceinfo_methods
+-- @libtbl surfaceinfo_meta
+
+SF.RegisterType("SurfaceInfo",false,true,debug.getregistry().SurfaceInfo)
+
+return function(instance) 
+
+local surfaceinfo_methods, surfaceinfo_meta, swrap, sunwrap = instance.Types.SurfaceInfo.Methods, instance.Types.SurfaceInfo, instance.Types.SurfaceInfo.Wrap, instance.Types.SurfaceInfo.Unwrap
+
+function surfaceinfo_meta:__tostring()
+	local surfinf = sunwrap(self)
+	if not surfinf then return "(null SurfaceInfo)"
+	else return tostring(surfinf) end
+end
+
+
+----------- Methods -----------
+
+local vec_meta, vwrap, vunwrap = instance.Types.Vector, instance.Types.Vector.Wrap, instance.Types.Vector.Unwrap
+
+--- Returns the brush surface's material.
+-- @shared
+-- @return Material of one portion of a brush model.
+function surfaceinfo_methods:getMaterial()
+    local surface = sunwrap(self)
+    return surface:GetMaterial()

```

Comment:
Has been done, had to set to client since server was giving weird errors which I should've expected since materials aren't really serverside.
Also got rid of the globals since they aren't being used for now

Reply:
Found a really weird bug https://imgur.com/a/RgeoGK6

Reply:
You shouldn't be able to set the flags of locked materials. They're locked so you can't modify them.

Reply:
I think it's a bug on our end so I'll fix that.

Reply:
The weird bug is because you modified that actual material the game was using for the map. Should be fixed now. What you want to do is create your own material and copy the texture.
`local mymaterial = material.create("VertexLitGeneric") mymaterial:setTexture("$basetexture",mat:getTexture("$basetexture"))`